python/itools-y4m.py

```diff
@@ -62,37 +55,34 @@ def do_range_conversion(inarr, srcmin, srcmax, dstmin, dstmax):
     # look for invalid values
     broken_range = False
     if (
-        len(outarr[outarr < np.iinfo(np.uint8).min]) > 0
-        or len(outarr[outarr > np.iinfo(np.uint8).max]) > 0
+        len(outarr[outarr < dstmin]) > 0
+        or len(outarr[outarr > dstmax]) > 0
     ):
         # strictly speaking, this y4m is wrong
         broken_range = True
+
     # clip values
-    outarr[outarr < np.iinfo(np.uint8).min] = np.iinfo(np.uint8).min
-    outarr[outarr > np.iinfo(np.uint8).max] = np.iinfo(np.uint8).max
-    outarr = outarr.astype(np.uint8)
+    outarr[outarr < dstmin] = dstmin
+    outarr[outarr > dstmax] = dstmax
+    outarr = outarr.astype(dt)
     return outarr, broken_range
 
+def get_colorspace_min_max(colorspace, colorrange):
+    if '10' in colorspace:
+        min = 0 if colorrange == "full" else 64
+        max = 1023 if colorrange == "full" else 940
+        return (min, max, np.uint16)
+    else:
+        min = 0 if colorrange == "full" else 16
+        max = 255 if colorrange == "full" else 235
```

Comment:
This is broken. Note that the limited-range definition is different for lumas and chromas. 
```
    - 8-bit color limited-range encoding
      - 8-bit color limited-range aka lr
      - Y' is typically quantized to the range [16, 235]
        - Y: [0x00, 0x0f]: other
          - Y=0: sync encoding
          - Y=1-15: overshoot
        - Y: [0x10, 0xeb]: other
          - Y=16: black
          - Y=17-234: grey
          - Y=235: white
        - Y: [0xec, 0xff]: other
          - Y=236-254: undershoot
          - Y=255: sync encoding
      - Cb/Cr are typically quantized to the range [16, 240]
        - Cb/Cr=128: zero value (used for encoding white, gray or black)
    - 10-bit color limited-range encoding
      - 10-bit color limited-range aka lr
      - Y: [64, 940]
      - Cb/Cr: [64, 960]
```

Reply:
I suggest to go back to the use of "luma_range_conversion()" and "chroma_range_conversion()"